allennlp/common/util.py

```diff
@@ -14,15 +14,17 @@
 
 JsonDict = Dict[str, Any]  # pylint: disable=invalid-name
 
-def sanitize(x: Any) -> Any:  # pylint: disable=invalid-name
+def sanitize(x: Any) -> Any:  # pylint: disable=invalid-name,too-many-return-statements
     """
     Sanitize turns PyTorch and Numpy types into basic Python types so they
     can be serialized into JSON.
     """
     if isinstance(x, (str, float, int, bool)):
         # x is already serializable
         return x
-    elif isinstance(x, torch.Tensor):
+    elif isinstance(x, torch.autograd.Variable):
+        return sanitize(x.data)
+    elif isinstance(x, (torch.Tensor, torch.LongTensor)):
```

Comment:
it might be good to just use `torch._TensorBase` here, b/c at some point someone will probably try to sanitize a `ByteTensor` or something and break it again.